### PR TITLE
add `let assert`

### PR DESCRIPTION
**Pull request by inoas**

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -440,7 +441,8 @@ module.exports = grammar({
       ),
     _case_clause_tuple_access: ($) =>
       seq(field("tuple", $.identifier), ".", field("index", $.integer)),
-    let: ($) => seq("let", $._assignment),
+		let_assert: ($) => seq("let assert", $._assignment),
+		let: ($) => seq("let", $._assignment),
```
Comment (the-mikedavis) [inline grammar.js:445]: looks like this switched to tabs: `npm run format` will fix the indentation

File: queries/highlights.scm [hunk 2]
```diff
@@ -77,6 +77,7 @@
   "fn"
   "if"
   "import"
+	"let assert"
```
Comment (the-mikedavis) [inline queries/highlights.scm:80]: Similar here - looks like this should be two spaces instead of a tab

File: test/corpus/functions.txt [hunk 3]
```diff
@@ -1082,5 +1136,5 @@ io.println("// hello world!\n")
     arguments: (arguments
       (argument
         value: (string
-	  (quoted_content)
-	  (escape_sequence))))))
+    (quoted_content)
+    (escape_sequence))))))
```
Comment (J3RN) [inline test/corpus/functions.txt:1140]: While we're talking about indentation, these two lines should not have changed :grin:

Comment (inoas) [reply test/corpus/functions.txt:1140]: these 2 lines had tabs and do not have tabs anymore.

**Review by J3RN — APPROVED** [review]
This looks great! :fire:  Once you rebase the branch I'll merge it :grin:

**Comment by inoas** [comment]
```
❯ ./scripts/integration_test.sh
Running parser against gleam-lang/stdlib
Total parses: 45; successful parses: 45; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/json
Total parses: 3; successful parses: 3; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/http
Total parses: 10; successful parses: 10; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/example-todomvc
Total parses: 23; successful parses: 23; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/bitwise
Total parses: 2; successful parses: 2; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/erlang
Total parses: 13; successful parses: 13; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/otp
Total parses: 13; successful parses: 13; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/cowboy
Total parses: 3; successful parses: 3; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/hackney
Total parses: 2; successful parses: 2; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/httpc
Total parses: 2; successful parses: 2; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/elli
Total parses: 4; successful parses: 4; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/javascript
Total parses: 7; successful parses: 7; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/example-echo-server
Total parses: 5; successful parses: 5; failed parses: 0; success percentage: 100.00%

Running parser against gleam-lang/plug
Total parses: 2; successful parses: 2; failed parses: 0; success percentage: 100.00%
```

File: grammar.js [hunk 4]
```diff
@@ -442,6 +443,7 @@ module.exports = grammar({
       ),
     _case_clause_tuple_access: ($) =>
       seq(field("tuple", $.identifier), ".", field("index", $.integer)),
+    let_assert: ($) => seq("let assert", $._assignment),
```
Comment (the-mikedavis) [inline grammar.js:446]: You may want to break this into two separate nodes:

```js
let_assert: ($) => seq("let", "assert", $._assignment),
```

Doing that allows tree-sitter "extras" to be inserted between the nodes, so you could have whitespace or comments between the `let` and `assert` tokens

Comment (inoas) [reply grammar.js:446]: not sure I would want to see that between those two though.
reason being is that in future assert alone will do something else than let assert, so:

```gleam
let 
// some comment
assert Ok(a) = Ok(1) // returns Ok(a) or crashes, "I think"
```

```gleam
assert Ok(a) = Ok(1) // returns true or crashes
```

Comment (inoas) [reply grammar.js:446]: But I am not certain.

Comment (inoas) [reply grammar.js:446]:
```gleam
    let
    // foo cool
    assert
    // foo cool
    Ok(string) =
      do_from_utf_codepoints_impl(utf_codepoints, bit_string.from_string(""))
      |> bit_string.to_string
    string

```
compiles though... the rest is the job of the formatter.

Comment (inoas) [reply grammar.js:446]: I have added the alt impl, the tests seem to pass so @the-mikedavis and @J3RN whatever you prefer. I think mike is correct here, not tree sitters decision if gleam allows the split.

**Comment by J3RN** [comment]
Closed in favor of #50

**Pull request closed without merging**